Tests/XcodeGenKitTests/SourceGeneratorTests.swift

```diff
@@ -605,19 +605,19 @@ class SourceGeneratorTests: XCTestCase {
                     TargetSource(path: "B", buildPhase: BuildPhaseSpec.none),
                     TargetSource(path: "C", buildPhase: nil),
                 ])
-                let project = Project(basePath: directoryPath, name: "Test", targets: [target])
+                let project = Project(basePath: directoryPath.absolute(), name: "Test", targets: [target])
```

Comment:
I had to change this to use `absolute()` here because otherwise the following logic wouldn't work:

https://github.com/yonaskolb/XcodeGen/blob/b8b457abdb29681b7692cf4148c299e213445d8d/Sources/XcodeGenKit/PBXProjGenerator.swift#L1178-L1180

Because `project.basePath` was just `TestDirectory/`, it couldn't work out the path relative to `plistPath` so instead was falling back to the absolute path. Is this just something that applied to this test case, or could we expect that `basePath` isn't always an absolute path?

I could move the `.absolute()` into my reimplementation of `getInfoPlists(for:)`, but since it wasn't already there, I don't know if I need to because there was already a bug here or because the `basePath` is already resolved elsewhere?

Reply:
I don't think we can rely on `basePath` being absolute, so maybe better to add to `getInfoPlists` if it relies on it. We could potentially make `basePath` always absolute, but I'm not sure if that will break anything else

Reply:
Ok thanks for confirming. I'll move the call to `absolute()` into `getInfoPlist(for:)` and try and add some tests around this to cover both scenarios